bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -961,4 +961,21 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_fips_mod() -> Result<(), Error> {
+        use crate::{enums::FipsMode, init};
+
+        init::init();
+
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "fips")] {
+                assert_eq!(init::fips_mode().unwrap(), FipsMode::Enabled);
+            } else {
+                assert_eq!(init::fips_mode().unwrap(), FipsMode::Disabled);
```

Comment:
If someone enables the fips feature flag in aws-lc-rs, but not in s2n-tls then I would consider that odd behavior. At that point this test might actually help them catch the discrepancy and either enable/disable fips for their application.

Reply:
Ah got it, okay. I guess I was thinking that someone might have aws-lc-rs as a direct dependency in addition to s2n-tls, or have multiple dependencies with an aws-lc-rs dependency, in which case just setting the fips flag for aws-lc-rs might make sense. But if this would be a strange thing to do, and they definitely should instead be setting the s2n-tls feature flag, then the test makes sense.